### PR TITLE
MCLOUD-7604: 'use_stale_cache' option use

### PR DESCRIPTION
**Pull request by dhorytskyi**
## Purpose of this pull request

Provide more information on how to use the "use_stale_cache" option.
Fix code samples.

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.4/config-guide/cache/two-level-cache.html
- https://devdocs.magento.com/cloud/env/variables-deploy.html

### CODE REVIEW

**Comment by dobooth** [comment]
Hi @dhorytskyi Please sign the Adobe CLA so we can process this PR.

**Comment by dhorytskyi** [comment]
Hi @dobooth I've signed it.

File: src/guides/v2.4/config-guide/cache/two-level-cache.md [hunk 1]
```diff
@@ -18,48 +18,52 @@ Magento stores the hashed data version in Redis, with the suffix ':version' appe
 
 ## Configuration example
 
+Put this into the `app/etc/env.php` file or replace existed cache section.
```
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/cache/two-level-cache.md:21]:
```suggestion
Put this into the `app/etc/env.php` file or replace the existing cache section.
```

File: src/guides/v2.4/config-guide/cache/two-level-cache.md [hunk 2]
```diff
@@ -18,48 +18,52 @@ Magento stores the hashed data version in Redis, with the suffix ':version' appe
...
--  `local_backend` is the local cache implementation: `Cm_Cache_Backend_File` or the APC adapter.
--  `cache_dir` is a directory where the local cache will be stored. It is suggested to use `/dev/shm/`.
--  `use_stale_cache` is a flag that enable\disables the stale cache mechanism.
+-  `backend` is the L2 cache implementation.
+-  `backend_options` are the L2 cache configuration.
+   -  `remote_backend` is the remote cache implementation: Redis or MySQL.
+   -  `remote_backend_options` are the remote cache configuration.
+   -  `local_backend` is the local cache implementation: `Cm_Cache_Backend_File` or the APC adapter.
+   -  `local_backend_options` are the local cache configuration.
+      -  `cache_dir` __a file cache specific option.__ Is a directory where the local cache will be stored.
```
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/cache/two-level-cache.md:63]:
```suggestion
      -  `cache_dir` __a file cache specific option.__ Is the directory where the local cache will be stored.
```

File: src/guides/v2.4/config-guide/cache/two-level-cache.md [hunk 3]
```diff
@@ -18,48 +18,52 @@ Magento stores the hashed data version in Redis, with the suffix ':version' appe
...
--  `backend` is the remote L2 cache implementation.
--  `remote_backend` is the remote cache implementation: Redis or MySQL.
--  `remote_backend_options` are Redis or MySQL-specific options.
--  `local_backend` is the local cache implementation: `Cm_Cache_Backend_File` or the APC adapter.
--  `cache_dir` is a directory where the local cache will be stored. It is suggested to use `/dev/shm/`.
--  `use_stale_cache` is a flag that enable\disables the stale cache mechanism.
+-  `backend` is the L2 cache implementation.
+-  `backend_options` are the L2 cache configuration.
+   -  `remote_backend` is the remote cache implementation: Redis or MySQL.
+   -  `remote_backend_options` are the remote cache configuration.
```
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/cache/two-level-cache.md:60]:
```suggestion
   -  `remote_backend_options` is the remote cache configuration.
```

File: src/guides/v2.4/config-guide/cache/two-level-cache.md [hunk 4]
```diff
@@ -18,48 +18,52 @@ Magento stores the hashed data version in Redis, with the suffix ':version' appe
...
 Where:
 
--  `backend` is the remote L2 cache implementation.
--  `remote_backend` is the remote cache implementation: Redis or MySQL.
--  `remote_backend_options` are Redis or MySQL-specific options.
--  `local_backend` is the local cache implementation: `Cm_Cache_Backend_File` or the APC adapter.
--  `cache_dir` is a directory where the local cache will be stored. It is suggested to use `/dev/shm/`.
--  `use_stale_cache` is a flag that enable\disables the stale cache mechanism.
+-  `backend` is the L2 cache implementation.
+-  `backend_options` are the L2 cache configuration.
```
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/cache/two-level-cache.md:58]:
```suggestion
-  `backend_options` is the L2 cache configuration.
```

File: src/guides/v2.4/config-guide/cache/two-level-cache.md [hunk 5]
```diff
@@ -18,48 +18,52 @@ Magento stores the hashed data version in Redis, with the suffix ':version' appe
...
+   -  `remote_backend` is the remote cache implementation: Redis or MySQL.
+   -  `remote_backend_options` are the remote cache configuration.
+   -  `local_backend` is the local cache implementation: `Cm_Cache_Backend_File` or the APC adapter.
+   -  `local_backend_options` are the local cache configuration.
```
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/cache/two-level-cache.md:62]:
```suggestion
   -  `local_backend_options` is the local cache configuration.
```

File: src/guides/v2.4/config-guide/cache/two-level-cache.md [hunk 6]
```diff
@@ -18,48 +18,52 @@ Magento stores the hashed data version in Redis, with the suffix ':version' appe
...
+      -  `cache_dir` __a file cache specific option.__ Is a directory where the local cache will be stored.
+   -  `use_stale_cache` is a flag that enable\disables the stale cache mechanism.
 
-We recommend the use of Redis for remote caching - `\Magento\Framework\Cache\Backend\Redis`, and the File cache implementation - `Cm_Cache_Backend_File` as the local cache.
+We recommend the use of Redis for remote caching - `\Magento\Framework\Cache\Backend\Redis`, and the File cache implementation - `Cm_Cache_Backend_File` as the local cache storing data in shared memory - `'local_backend_options' => ['cache_dir' => '/dev/shm/']`.
```
Comment (BarnyShergold) [inline src/guides/v2.4/config-guide/cache/two-level-cache.md:66]:
```suggestion
We recommend the use of Redis for remote caching - `\Magento\Framework\Cache\Backend\Redis`, and the File cache implementation - `Cm_Cache_Backend_File` as the local cache for storing data in shared memory - `'local_backend_options' => ['cache_dir' => '/dev/shm/']`.
```

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Nice update - just some grammatical corrections

**Comment by dhorytskyi** [comment]
@BarnyShergold Thank you!

**Review by BarnyShergold — APPROVED** [review]

**Review by oshmyheliuk — APPROVED** [review]

File: src/guides/v2.4/config-guide/cache/two-level-cache.md [hunk 7]
```diff
@@ -18,48 +18,52 @@ Magento stores the hashed data version in Redis, with the suffix ':version' appe
...
+-  `backend_options` is the L2 cache configuration.
+   -  `remote_backend` is the remote cache implementation: Redis or MySQL.
+   -  `remote_backend_options` is the remote cache configuration.
+   -  `local_backend` is the local cache implementation: `Cm_Cache_Backend_File` or the APC adapter.
+   -  `local_backend_options` is the local cache configuration.
+      -  `cache_dir` __a file cache specific option.__ Is the directory where the local cache will be stored.
+   -  `use_stale_cache` is a flag that enable\disables the stale cache mechanism.
 
-We recommend the use of Redis for remote caching - `\Magento\Framework\Cache\Backend\Redis`, and the File cache implementation - `Cm_Cache_Backend_File` as the local cache.
+We recommend the use of Redis for remote caching - `\Magento\Framework\Cache\Backend\Redis`, and the File cache implementation - `Cm_Cache_Backend_File` as the local cache for storing data in shared memory - `'local_backend_options' => ['cache_dir' => '/dev/shm/']`.
```
Comment (bdenham) [inline src/guides/v2.4/config-guide/cache/two-level-cache.md:66]:
```suggestion
We recommend using Redis for remote caching (`\Magento\Framework\Cache\Backend\Redis`) and the `Cm_Cache_Backend_File` implementation for local caching of data in shared memory, using `'local_backend_options' => ['cache_dir' => '/dev/shm/']`.
```

**Review by bdenham — CHANGES_REQUESTED** [review]
Just a few wording changes. Thanks @dhorytskyi.

**Comment by bdenham** [comment]
Hey @dhorytskyi, just following up with you on this PR. Its ready to merge after you make the wording changes suggested above.

**Comment by dhorytskyi** [comment]
@bdenham Thank you, I've applied your suggestions.

**Comment by bdenham** [comment]
running tests

**Comment by bdenham** [comment]
@magento import code to magento-devdocs/devdocs

**Comment by magento-engcom-team** [comment]
@bdenham the branch with code successfully imported into`magento-devdocs/devdocs` repository. Branch name: `imported-magento-devdocs-8672`.

**Review by bdenham — APPROVED** [review]

**Pull request merged**